src/services/ledger/index.types.d.ts

```diff
@@ -0,0 +1,37 @@
+type TxMetadata = {
+  type: LedgerTransactionType
+  pending: boolean
+  fee: Satoshis
+  feeUsd: number
+  sats: Satoshis
+  usd: number
+  currency: Currency
+}
+
+type AddLnTxReceiveMetadata = TxMetadata & {
+  hash: PaymentHash
+}
+
+type AddLnTxSendMetadata = TxMetadata & {
+  hash: PaymentHash
+  pubkey: Pubkey
+  feeKnownInAdvance: boolean
+}
+
+type AddLnIntraledgerTxSendMetadata = TxMetadata & {
+  hash: PaymentHash
+  pubkey: Pubkey
+  memoPayer: string | null
+  username: WalletName | null
+}
+
+type AddOnChainIntraledgerTxSendMetadata = TxMetadata & {
+  payee_addresses: OnChainAddress[]
+  sendAll: boolean
+  memoPayer: string | null
```

Comment:
dont we use `memoPayer?: string` because we need to persist `memoPayer` attribute? or because we need null instead undefined?

Reply:
We've been discussing how to represent the absence of an expected value like this (`undefined`, `""`, or `null`) and I believe the general preference is to use `null` wherever possible. @bodymindarts or @samerbuna would have to confirm